chapters/memory-layout/memory-operations/reading/README.md

```diff
@@ -0,0 +1,368 @@
+# Pointers
+
+In the C programming language, memory interaction is achieved through pointers.
+We remind you that a pointer is a variable that holds a memory address.
+The general declaration form is as follows: `type *variable_name`, where `variable_name` can represent any valid data type in C.
+
+> **WARNING:** The **asterisk (`*`)** used in declaring a pointer denotes that it is a pointer and should not be confused with the dereference operator.
+These are two entirely different concepts represented by the same symbol.
+> Declaring a pointer does not mean allocating a memory area to store data.
+A pointer is also a data type, whose value is a number representing a memory address.
+The size of the pointer data type is always the same, regardless of the type of data it points to, and is determined by the architecture and operating system on which the program was compiled (but usually 4 bytes on 32-bit systems and 8 bytes on 64-bit systems).
+>
+> ```c
+>   int *p = 0xCAFEBABE; /* Declaring a pointer */
+>   int x = *p; /* The value at the address stored in p */
+> ```
+
+In C, a pointer can represent:
+
+- The address of data of a certain type
+- The address of a memory area
+- The address of a function
+- The address of an area with unknown content (void pointer)
```

Comment:
This is a bit confusing. Normally, a pointer holds a memory address, be it data or code (functions). And the void pointer doesn't hold the address of an area with unknown content, rather it holds the address where data of unknown type is held.